### PR TITLE
CNF-8039: Add PreCachingConfig CRD

### PR DESCRIPTION
**Pull request by sakhoury**
This PR contains the PreCachingConfig custom resource definition as described in https://github.com/openshift-kni/kni-enhancements/blob/main/enhancements/cgu-precaching-additional-images.md.

/cc @jc-rh @sudomakeinstall2 @vitus133 @irinamihai

### CODE REVIEW

File: api/v1alpha1/precachingconfig_types.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// Overrides modify the default pre-caching behavior and values dervied by TALM.
+type Overrides struct {
```
[the remark this answers is not in the document]
Comment (jc-rh) [reply api/v1alpha1/precachingconfig_types.go]: Please update the PrecachingSpec structure in clustergroupupgrade_types.go as well.

[the remark this answers is not in the document]
Comment (sakhoury) [reply api/v1alpha1/precachingconfig_types.go]: This change will be addressed in a subsequent PR when the clustergroupupgrade codebase will be updated with added functionality for additional user images field.

**Pull request merged**